### PR TITLE
eliminate zone_add create_resources, add forwarder and /24 support

### PR DESCRIPTION
**Pull request by adamcstephens**
* Turn hiera lookups into parameters, using type validation.
* Replace zone_add defined type with an iterator
* Add option for disabling IPAM lookup, which will provide a fake set of zone data to the module
* Add support for forwarder zones
* Disable DNSSEC by default since bind cannot toggle DNSSEC per-zone
* Fix /24 support

### CODE REVIEW

**Comment by ryanbsaunders** [comment]
Looks 🆗 to me, but someone who has worked more closely with our BIND infrastructure may want to weigh in as well.

File: lib/puppet/parser/functions/dns_array.rb [hunk 1]
```diff
@@ -21,17 +22,51 @@ module Puppet::Parser::Functions
...
+          http = Net::HTTP.new(uri.host, uri.port)
+          http.use_ssl = true
+          http.ssl_version=:TLSv1_2
+          http.verify_mode = OpenSSL::SSL::VERIFY_NONE
+          http.read_timeout = 30
+          request = Net::HTTP::Get.new(uri.request_uri)
+          response = http.request(request)
+          response.body
+        end
+      else
```
Comment (dugdigg) [inline lib/puppet/parser/functions/dns_array.rb]: I would rather that the default for false would be not to use the remote source for data.  Leave a small example of stubbing out fake addresses to allow someone to enable it if they needed to.

Comment (adamcstephens) [inline lib/puppet/parser/functions/dns_array.rb]: Huh? The default is to use ipam. Maybe I'm misunderstanding you?

**Comment by adamcstephens** [comment]
I'm splitting this into multiple PRs

**Pull request closed without merging**